internal/resources/policy/type/custom/resource_custom_policy.go

```diff
@@ -0,0 +1,421 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package custom
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	objectmetamodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/objectmeta"
+	policymodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy"
+	policyclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/cluster"
+	policyclustergroupmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/clustergroup"
+	policyorganizationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/organization"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy"
+)
+
+func ResourceCustomPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCustomPolicyCreate,
+		ReadContext:   resourceCustomPolicyRead,
+		UpdateContext: resourceCustomPolicyInPlaceUpdate,
+		DeleteContext: resourceCustomPolicyDelete,
+		Schema:        customPolicySchema,
+		CustomizeDiff: customdiff.All(
+			policy.ValidateScope,
+			validateInput,
+			policy.ValidateSpecLabelSelectorRequirement,
+		),
+	}
+}
+
+var customPolicySchema = map[string]*schema.Schema{
+	nameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the custom policy",
+		Required:    true,
+		ForceNew:    true,
+	},
+	policy.ScopeKey: policy.ScopeSchema,
+	common.MetaKey:  common.Meta,
+	specKey:         specSchema,
+}
+
+func resourceCustomPolicyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	customPolicyName, _ := d.Get(nameKey).(string)
+
+	scopedFullnameData := policy.ConstructScope(d, customPolicyName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control cluster custom policy entry; Scope full name is empty")
+	}
```

Comment:
Please try to reuse these error formats across security and custom policies by declaring them at the policy level.